### PR TITLE
add country math feature per #34, update missing docs and release version 1.3

### PR DESCRIPTION
**Pull request by Ths2-9Y-LqJt6**
- add feature per #34 and add new signature to `map.countries.fill.transform` docs
- fix docs: missing `map.markers.rollup`, `map.markers.groupby`, `map.markers.attr.r.rollup`, `map.markers.attr.r.groupby`, `map.markers.attr.r.min`, `map.markers.attr.r.max` and  `map.countries.attr.fill.transform` 
- `map.countries.attr.fill.rollup` is in the wrong spot in docs

### CODE REVIEW

**Comment by Ths2-9Y-LqJt6** [comment]
@melalj - at your leisure for a review!

File: src/components/GeoMap.js [hunk 1]
```diff
@@ -473,7 +472,7 @@ export default class GeoMap {
         } else {
           const originalValueRaw = d.rollupValue[attrKey];
           const originalValue = (attrValue.transform) ?
-              attrValue.transform(originalValueRaw) : originalValueRaw;
+              attrValue.transform(originalValueRaw, this.maptable.rawData) : originalValueRaw;
```
Comment (Ths2-9Y-LqJt6) [inline src/components/GeoMap.js]: this is exactly what #34 is asking for, more data! by passing both the ``val`` and ``this.maptable.rawData`` you can do comparative functions, like percentile.

Comment (melalj) [inline src/components/GeoMap.js]: You should document it on the README.
But here you're passing the whole raw dataset as parameter, and not the raw row

Comment (melalj) [inline src/components/GeoMap.js]: Oh spoke to quickly, that's useful indeed to calculate the percentile. Yep Maybe you should add an example to understand why we're passing the raw dataset :)

Comment (Ths2-9Y-LqJt6) [inline src/components/GeoMap.js]: good call on the README update - i'll add that to this PR, stay tuned!

Comment (Ths2-9Y-LqJt6) [inline src/components/GeoMap.js]: @melalj  - okied dokie!  docs are updated.  i got a little turned around with all the different groupby, transform and rollup declarations so lemme know if they don't look right to you!

File: README.md [hunk 2]
```diff
@@ -1,7 +1,7 @@
 MapTable
 ========
 
-[![GitHub stars](https://img.shields.io/github/stars/Packet-Clearing-House/maptable.svg?style=social&label=Star&maxAge=2592001)]() [![GitHub release](https://img.shields.io/github/release/Packet-Clearing-House/maptable.svg?maxAge=2592002)]() [![license](https://img.shields.io/github/license/Packet-Clearing-House/maptable.svg?maxAge=2592001)]()
+[![GitHub stars](https://img.shields.io/github/stars/Packet-Clearing-House/maptable.svg?style=social&label=Star&maxAge=2592003)]() [![GitHub release](https://img.shields.io/github/release/Packet-Clearing-House/maptable.svg?maxAge=2592003)]() [![license](https://img.shields.io/github/license/Packet-Clearing-House/maptable.svg?maxAge=2592004)]()
```
Comment (melalj) [inline README.md]: What is it for?

File: README.md [hunk 3]
```diff
@@ -333,11 +333,16 @@ watermark: {
...
+    - `markers.rollup:` _(function(groupedData))_ Given groupedData (the current row), function that returns a value that we would use for every marker.
     - `markers.tooltipClassName:` _(string, default: 'mt-map-tooltip popover bottom')_ Class name of the tooltip used for markers (we're using bootstrap).
     - `markers.tooltip:` _(function(groupedData))_ Function that returns html that we would use as content for the tooltip. We recommend you to use the bootstrap popover..
     - `markers.attr:` _(object)_ Markers attributes (same naming as SVG attributes).
         - `markers.attr.fill:` _(ScaledValue)_ Marker background color.
-        - `markers.attr.r:` _(ScaledValue)_ Marker radius.
+        - `markers.attr.r.max:` _(ScaledValue)_ Maximum radius.
+        - `markers.attr.r.min:` _(ScaledValue)_ Minimum radius
+        - `markers.attr.r.rollup:` _(function(groupedData), default: values.length)_ Function for the values we're attaching to the radius. Defaults to ``values.length``
+        - `markers.attr.r.transform:` _(function(value), default: value)_ Function for changing the value for radius of the marker (eg ``Math.log(value)``)
```
Comment (melalj) [inline README.md]: `function(value, rawDataset)`

File: examples/negative.example.html [hunk 4]
```diff
@@ -28,6 +28,9 @@
                             maxNegative: "tomato",
                             empty: "#f9f9f9",
                             legend: true,
+                            transform: function(val) {
```
Comment (melalj) [inline examples/negative.example.html]: this function seems useless here

File: src/components/GeoMap.js (hunk 1 above)
Comment (melalj) [inline src/components/GeoMap.js]: Do you have any example on how you calculate the percentile? Maybe it can be interesting to see its application in the example folder

File: README.md (hunk 2 above)
Comment (Ths2-9Y-LqJt6) [inline README.md]: They're cached and don't update unless we change the URL.

File: README.md (hunk 3 above)
Comment (Ths2-9Y-LqJt6) [inline README.md]: I don't follow.  Did you mean this for line 411 instead? That's:

```
`countries.attr.transform:` _(function(value, allRows), default: value)_
```

you want me to chant that to this maybe?

```
`countries.attr.transform:` _(function(value, rawDataset), default: value)_
``

File: examples/negative.example.html (hunk 4 above)
Comment (Ths2-9Y-LqJt6) [inline examples/negative.example.html]: ah - sure. see f3de568

File: src/components/GeoMap.js (hunk 1 above)
Comment (Ths2-9Y-LqJt6) [inline src/components/GeoMap.js]: yeah...i thought about adding this somewhere. What I'd like to do is actually add an EXAMPLES.md to the parent directory.  Then yanks all the examples of more than a 1 line that break up the README.md and add deep links to EXAMPLES.md.  Sound good?  I'd do this as a separate PR than this one.

File: README.md (hunk 3 above)
Comment (melalj) [inline README.md]: event markers have a `.transform` property. It should use the same signature as the one for countries

File: README.md (hunk 2 above)
Comment (melalj) [inline README.md]: ok

File: README.md (hunk 3 above)
Comment (Ths2-9Y-LqJt6) [inline README.md]: Sorry, I'm still confused :(

I know ``markers.attr.r`` has ``markers.attr.r.transform`` property as it was missing and I added it ;)  However, it can not take the same signature as ``countries.attr.transform`` because that accepts the value *and* rawDataset.  The signature for ``markers.attr.r.transform`` is still just the ``value``.   Maybe we need to chat real time to see what you're getting at? I wanna fix it!!

Comment (melalj) [inline README.md]: You've added the `transform` on the `setAttrValues(` that is used for for all dynamic attr values (countries, markers). https://github.com/Packet-Clearing-House/maptable/blob/dev-34/src/components/GeoMap.js#L475

So if you change the behavior of `setAttrValues` it will affect both countries and markers, and thus the transform functions can apply to any attribute of countries or markers.

Let me know if it not clear, we can switch to live chat.

File: src/components/GeoMap.js (hunk 1 above)
Comment (melalj) [inline src/components/GeoMap.js]: ok

File: README.md (hunk 3 above)
Comment (Ths2-9Y-LqJt6) [inline README.md]: Oooooooh, right!  gotcha.  fixed in 2ada14f.  Thanks for walking me through that!

File: src/components/GeoMap.js (hunk 1 above)
Comment (Ths2-9Y-LqJt6) [inline src/components/GeoMap.js]: Awesome! I spun up #36 to cover that.  stay tuned!

File: README.md (hunk 3 above)
Comment (melalj) [inline README.md]: One more thing:
attr actually can take any SVG attribute (r, fill, stroke, stroke-with, opacity, x, y ...), and every attribute can be either a string/integer to se a constant value or be an object (min, max, rollup, transform..) to make it dynamic 😉

Comment (Ths2-9Y-LqJt6) [inline README.md]: OK...but now I'm back to being lost again :(

Do you mean that we need to enumerate more of options in our README.md docs to cover items that I'm missing for attr calls to the SVG?  Or do you mean that we need to update the way we define the signatures to call out that you can pass in both a constant or an object?

Comment (melalj) [inline README.md]: That would be useful if someone is looking to expand the MapTable to different use cases.
Let not enumerate all of them but just say that we can pass either a constant (for static value) or object (for dynamic value)

**Comment by Ths2-9Y-LqJt6** [comment]
@melalj  - I realize we never merged this PR - OK to merge it as is and then we can come back and tidy up the docs per #36?
